File: NutriSmart/app/src/main/java/com/example/nutrismart/data/repository/FastingRepository.kt
package com.example.nutrismart.data.repository

import com.example.nutrismart.data.SessionManager
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class FastingRepository @Inject constructor(
    private val sessionManager: SessionManager
) {
    fun isFastingActive(): Boolean = sessionManager.isFastingActive()
    
    fun getFastingEndTime(): Long = sessionManager.getFastingEndTime()
    
    fun getFastingStartTime(): Long = sessionManager.getFastingStartTime()
    
    fun getFastingDurationHours(): Int = sessionManager.getFastingDurationHours()
    
    fun clearFastingState() = sessionManager.clearFastingState()
    
    fun saveFastingState(isActive: Boolean, startTime: Long, endTime: Long, durationHours: Int) {
        sessionManager.saveFastingState(isActive, startTime, endTime, durationHours)
    }
}

File: NutriSmartIOS/data/repository/FastingRepository.swift
//
//  FastingRepository.swift
//  NutriSmartIOS
//

import Foundation

class FastingRepository {
    static let shared = FastingRepository()
    private let sessionManager = SessionManager.shared
    
    private init() {}
    
    func isFastingActive() -> Bool { sessionManager.isFastingActive() }
    func getFastingEndTime() -> Date {
        let time = sessionManager.getFastingEndTime()
        return Date(timeIntervalSince1970: TimeInterval(time))
    }
    
    func getFastingStartTime() -> Date {
        let time = sessionManager.getFastingStartTime()
        return Date(timeIntervalSince1970: TimeInterval(time))
    }
    
    func getFastingDurationHours() -> Int { sessionManager.getFastingDurationHours() }
    func clearFastingState() { sessionManager.clearFastingState() }
    
    func saveFastingState(isActive: Bool, startTime: Date, endTime: Date, durationHours: Int) {
        sessionManager.saveFastingState(
            isFasting: isActive,
            startTime: Int(startTime.timeIntervalSince1970),
            endTime: Int(endTime.timeIntervalSince1970),
            durationHours: durationHours
        )
    }
}
